### DISCUSSION TITLE
Is it possible to expand/deflate aliases?

### DISCUSSION THREAD

**Discussion by telometto** [akinomyoga/ble.sh#260] [Q&A] [answered]
E.g. if `alias ls='ls --color=auto'`, is it possible to have `ble.sh` expand the full command if I only typed `ls` or `\ls` and hit <kbd>Space</kbd>, just like with the `\date` recipe?

**Comment by akinomyoga** [comment]
Thank you! Currently, there is no such feature in ble.sh, but it is actually in a ToDo list (note.txt:3946). I'll make its priority higher.

**Comment by akinomyoga** [comment, marked as answer]
I have newly implemented [`bleopt edit_magic_expand`](https://github.com/akinomyoga/ble.sh/wiki/Manual-%C2%A74-Editing#user-content-bleopt-edit_magic_expand) in commit 63da2ac. You can use the following setting to enable all the history expansion, the sabbrev expansion, and the alias expansion by `magic-space` (the default <kbd>SP</kbd> binding):

```bash
# blerc

bleopt edit_magic_expand=history:sabbrev:alias
```

Could you update `ble.sh` and check its behavior? Thank you!

**Reply by telometto** [reply]
Works immaculately! As always, amazing work 🥳

**Reply by akinomyoga** [reply]
Thanks!
